### PR TITLE
Add "import/export without validating" button

### PR DESCRIPTION
**Pull request by lacardonap**
When you import/export data and an error occurs, a new button is enabled that allows you to execute data import/export using `--disableValidation`.

We did our best trying to isolate errors that come from other sources, in fact, Model Baker already handles JAVA-related issues (no Java installed/given, no proper version) and connection issues very well.

init_ui_import_data

import_with_validating

import_with_disable_validation

References https://github.com/opengisch/QgisModelBaker/issues/318

### CODE REVIEW

File: QgisModelBaker/gui/import_data.py [hunk 1]
```diff
@@ -291,7 +336,9 @@ def enable(self):
         self.buttonBox.setEnabled(True)
 
     def type_changed(self):
+        self.remove_import_without_validate_button()
         self.progress_bar.hide()
+        self.txtStdout.clear()
```
Comment (m-kuhn) [inline QgisModelBaker/gui/import_data.py]: Is that related to this change? Could this be done in a separate pull request?

File: QgisModelBaker/gui/import_data.py [hunk 2]
```diff
@@ -122,12 +122,24 @@ def __init__(self, base_config, parent=None):
...
+        self.xtf_file_line_edit.textChanged.connect(
+            self.remove_import_without_validate_button)
+
         settings = QSettings()
         ilifile = settings.value('QgisModelBaker/ili2db/ilifile')
         self.ilicache = IliCache(base_config, ilifile or None)
         self.update_models_completer()
         self.ilicache.refresh()
 
+    def accepted_import_data(self, button):
```
[the remark this answers is not in the document]
Comment (lacardonap) [reply QgisModelBaker/gui/import_data.py]: Suggested changes were made in commit 62acfd2

File: QgisModelBaker/gui/import_data.py [hunk 3]
```diff
@@ -122,12 +122,24 @@ def __init__(self, base_config, parent=None):
         self.xtf_file_line_edit.textChanged.emit(
             self.xtf_file_line_edit.text())
 
+        # Remove import without validate button when xtf change
```
[the remark this answers is not in the document]
Comment (lacardonap) [reply QgisModelBaker/gui/import_data.py]: Suggested changes were made in commit 62acfd2

File: QgisModelBaker/gui/export.py [hunk 4]
```diff
@@ -298,8 +308,36 @@ def on_process_finished(self, exit_code, result):
...
+            if self.export_without_validate():
+
+                # button is removed to define order in GUI
+                for button in self.buttonBox.buttons():
+                    if button.text() == self.tr('Export'):
+                        self.buttonBox.removeButton(button)
+                # Check if button was previously added
+                self.remove_export_without_validate_button()
+
+                self.buttonBox.addButton(self.tr('Export without validating'),
```
[the remark this answers is not in the document]
Comment (lacardonap) [reply QgisModelBaker/gui/export.py]: Suggested changes were made in commit 62acfd2

File: QgisModelBaker/gui/export.py [hunk 5]
```diff
@@ -298,8 +308,36 @@ def on_process_finished(self, exit_code, result):
...
+        :return: True if you can execute the export without validations, False in other case
+        """
+        log = self.txtStdout.toPlainText().lower()
+        if "permission denied" in log or "access is denied" in log:
+            return False
+        return True
+
+    def remove_export_without_validate_button(self):
+        for button in self.buttonBox.buttons():
+            if button.text() == self.tr('Export without validating'):
```
[the remark this answers is not in the document]
Comment (lacardonap) [reply QgisModelBaker/gui/export.py]: It is not possible because two buttons have the same role. Now the name of the button is assigned in a single point to avoid typos errors.

File: QgisModelBaker/gui/import_data.py (hunk 1 above)
Comment (lacardonap) [reply QgisModelBaker/gui/import_data.py]: yes, the change was made in the PR #334

File: QgisModelBaker/gui/export.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (m-kuhn) [reply QgisModelBaker/gui/export.py]: I thought we could assign a different role to the export without validation one?
But anyway, the current solution is much better.
